core/scripts/server.ts

```diff
@@ -194,6 +194,11 @@ app.get('*', (req, res, next) => {
     }
     const context = ssr.initSSRRequestContext(app, req, res, config)
     renderer.renderToString(context).then(output => {
+      if (context.server._redirect.isPending()) {
```

Comment:
It looks good. However, what would happen if we want to redirect the user from the cached page? I think we should also support this case

Reply:
In that case you need to clear cache. `dynamicRequestHandler` will not be resolved for cached pages. So if you changed urls, you need to clear cache and apply logic in asyncData that will handle redirection. I don't see other way :thinking:

Reply:
Couldn't we apply a similar mechanism to dynamicCacheHandler?